stock_request/models/stock_request.py

```diff
@@ -340,6 +341,49 @@ def _prepare_procurement_values(self, group_id=False):
     def _skip_procurement(self):
         return self.state != "draft" or self.product_id.type not in ("consu", "product")
 
+    def _prepare_stock_move(self):
+        return {
+            "name": self.product_id.display_name,
+            "company_id": self.company_id.id,
+            "product_id": self.product_id.id,
+            "product_uom_qty": self.product_uom_qty,
+            "product_uom": self.product_id.uom_id.id,
+            "location_id": self.warehouse_id.lot_stock_id.id,
+            "location_dest_id": self.location_id.id,
+            "state": "draft",
+            "reference": self.name,
+        }
+
+    def _prepare_stock_request_allocation(self, move):
+        return {
+            "stock_request_id": self.id,
+            "stock_move_id": move.id,
+            "requested_product_uom_qty": self.product_uom_qty,
+        }
+
+    def _action_stock_move_process(self):
+        move = self.env["stock.move"].create(self._prepare_stock_move())
+        move._action_confirm()
+        move._action_assign()
+        move.quantity_done = move.product_uom_qty
+        move._action_done()
+        return move
+
+    def _action_use_stock_available(self):
+        allocation_model = self.env["stock.request.allocation"]
+        stock_move_model = self.env["stock.move"].sudo()
+        for request in self.filtered(
+            lambda x: x.company_id.stock_request_check_available_stock_first
+        ):
+            if request.product_id.sudo().qty_available >= request.product_uom_qty:
```

Comment:
You are not passing location as context for restricting the available quantity. I also understand that you need to take into account other SR allocations for reducing this quantity.